controller/src/main/java/gov/va/vro/controller/VroController.java

```diff
@@ -99,6 +99,44 @@ public ResponseEntity<Object> fetchPdf(String claimSubmissionId) throws ClaimPro
     }
   }
 
+  @Override
+  public ResponseEntity<Object> generateFetchPdf(GeneratePdfRequest request)
+      throws ClaimProcessingException {
+    log.info(
+        "Generating pdf for claim: {} and diagnostic code {}",
+        request.getClaimSubmissionId(),
+        request.getDiagnosticCode());
+    try {
+      GeneratePdfPayload model = generatePdfRequestMapper.toModel(request);
+      log.info(model.toString());
+      String response = camelEntrance.generateFetchPdf(model);
+      FetchPdfResponse pdfResponse = objectMapper.readValue(response, FetchPdfResponse.class);
+      log.info("RESPONSE from fetchPdf returned status: {}", pdfResponse.getStatus());
```

Comment:
Can we move commonalities with the existing endpoints to common methods?

Reply:
Cleaned up into a `fetchPdfProcess` function that gets used for the fetch and immediate pdf routes